packages/delete/src/modules/delete/sagas.js

```diff
@@ -0,0 +1,78 @@
+import {takeLatest, all, call, put, select} from 'redux-saga/effects'
+import {rest, externalEvents} from 'tocco-app-extensions'
+import {selection as selectionUtil} from 'tocco-util'
+
+import * as actions from './actions'
+import deleteRequestParser from '../../utils/deleteRequestParser'
+
+export const inputSelector = state => state.input
+export const textResourceSelector = state => state.intl.messages
+
+export function* getDeleteBodyFromSelection() {
+  const {selection} = yield select(inputSelector)
+  const entities = yield call(selectionUtil.getEntities, selection, rest.fetchEntities)
+
+  return {
+    entityModel: entities.entityName,
+    keys: entities.keys
+  }
+}
+
+const deleteDialogResource = 'client/delete/dialog'
+
+export function* loadDialogInfo() {
+  const body = yield call(getDeleteBodyFromSelection)
+
+  const response = yield call(rest.requestSaga, deleteDialogResource, {method: 'POST', body})
+  const principal = yield call(rest.fetchPrincipal)
```

Comment:
could `requestSaga` and `fetchPrincipal` also be executed in parallel? seems like they don't depend on each other.